src/script/lua_api/l_object.cpp

```diff
@@ -738,6 +738,28 @@ int ObjectRef::l_get_attach(lua_State *L)
 	return 5;
 }
 
+// get_children(self)
+int ObjectRef::l_get_children(lua_State *L)
+{
+	GET_ENV_PTR;
+
+	ObjectRef *ref = checkobject(L, 1);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == NULL)
+		return 0;
+
+	//Do it
+	const std::unordered_set<int> child_ids = sao->getAttachmentChildIds();
+	int i = 0;
+	lua_createtable(L, child_ids.size(), 0);
+	for (const int &id : child_ids) {
```

Comment:
reference doesn't make much sense here
```suggestion
	for (int id : child_ids) {
```

Reply:
You are advising the opposite of what the c++ core guidelines say
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#S-const

Why not use `auto const &id`? There's not reason not to use const reference

Reply:
> even better:
Why? You increased the complexity, worsened readability, and added a bug, for literally no gain

Reply:
A reference is (almost universally) internally a pointer, i.e. 4 or 8 bytes.  An `int` is 4 bytes.  So, what the point of a reference?  There's no way a reference will be faster.  `const` can be kept, but a reference is superfluous.

Obviously, it doesn't matter much.  But for the sake of the code guidelines, "Don't be too C++y", I say no to a reference in this specific instance.

Reply:
reference vs value doesn't make a difference on optimized gcc, and I'm not even sure any optimization is needed since this is not a function call. But ok, fair enough

There's still no reason not to use const

Reply:
`const` or not is a style question, I consider it unnecessary for such a small loop body.

A reference here is nonsense, as already explained.